src/checkers/inference/model/ConstraintManager.java

```diff
@@ -259,4 +263,8 @@ public void addArithmeticConstraint(ArithmeticOperationKind operation, Slot left
             Slot rightOperand, ArithmeticVariableSlot result) {
         add(createArithmeticConstraint(operation, leftOperand, rightOperand, result));
     }
+
+    public void addImplicationConstraint(List<Constraint> assumptions, Constraint conclusion) {
+        add(createImplicationConstraint(assumptions, conclusion));
```

Comment:
If all assumptions are normalized to true, and conclusion is normalized to false, the result is normalized to false. Do you need to issue a custom error for this case?

Reply:
No, it just leverages the error message inside `add` method